.devcontainer/Dockerfile

```diff
@@ -0,0 +1,189 @@
+RUN make -C .pgenv/src/postgresql-*/src/include install
+
+# create a staging directory with all files we want to copy from our pgenv build
+# we will copy the contents of the staged folder into the final image at once
+RUN mkdir .pgenv-staging/
+RUN cp -r .pgenv/src .pgenv/pgsql-* .pgenv/config .pgenv-staging/
+RUN rm .pgenv-staging/config/default.conf
+
+FROM base AS pg15
+RUN pgenv build 15.4
+RUN rm .pgenv/src/*.tar*
+RUN make -C .pgenv/src/postgresql-*/ clean
+RUN make -C .pgenv/src/postgresql-*/src/include install
+
+# create a staging directory with all files we want to copy from our pgenv build
+# we will copy the contents of the staged folder into the final image at once
+RUN mkdir .pgenv-staging/
+RUN cp -r .pgenv/src .pgenv/pgsql-* .pgenv/config .pgenv-staging/
+RUN rm .pgenv-staging/config/default.conf
+
+FROM base AS pg16
+RUN pgenv build 16rc1
+RUN rm .pgenv/src/*.tar*
+RUN make -C .pgenv/src/postgresql-*/ clean
+RUN make -C .pgenv/src/postgresql-*/src/include install
+
+# create a staging directory with all files we want to copy from our pgenv build
+# we will copy the contents of the staged folder into the final image at once
+RUN mkdir .pgenv-staging/
+RUN cp -r .pgenv/src .pgenv/pgsql-* .pgenv/config .pgenv-staging/
+RUN rm .pgenv-staging/config/default.conf
+
+FROM base AS uncrustify-builder
+
+RUN sudo apt update && sudo apt install -y cmake tree
+
+WORKDIR /uncrustify
+RUN curl -L https://github.com/uncrustify/uncrustify/archive/uncrustify-0.68.1.tar.gz | tar xz
+WORKDIR /uncrustify/uncrustify-uncrustify-0.68.1/
+RUN mkdir build
+WORKDIR /uncrustify/uncrustify-uncrustify-0.68.1/build/
+RUN cmake ..
+RUN make -sj8
+
+RUN make install DESTDIR=/uncrustify
+
+FROM base AS pipenv
+
+RUN sudo pip3 install pipenv pipenv-shebang
+
+WORKDIR /workspaces/citus/
+COPY src/ src/
+COPY .vscode/Pipfile .vscode/Pipfile.lock .devcontainer/.vscode/
+RUN ( cd src/test/regress && pipenv install )
+RUN ( cd .devcontainer/.vscode && pipenv install )
+
+# assemble the final container by copying over the artifacts from separately build containers
+FROM base AS devcontainer
+
+LABEL org.opencontainers.image.source=https://github.com/citusdata/citus
+LABEL org.opencontainers.image.description="Development container for the Citus project"
+LABEL org.opencontainers.image.licenses=AGPL-3.0-only
+
+RUN yes | sudo unminimize
+
+# install developer productivity tools
+RUN sudo apt update \
+ && sudo apt install -y \
+    autoconf2.69 \
+    bash-completion \
+    fswatch \
+    gdb \
+    htop \
+    libdbd-pg-perl \
+    libdbi-perl \
+    lsof \
+    man \
+    net-tools \
+    pspg \
+    vim \
+ && sudo apt clean
+
+RUN sudo pip3 install pipenv pipenv-shebang
+
+# add some common tools to the final container
+# bin directory for user tools
+RUN mkdir .bin
+ENV PATH="/home/citus/.bin:${PATH}"
+
+# install citus-dev
+RUN git clone --branch feature/citus-dev-pipenv-shebang https://github.com/citusdata/tools.git citus-tools \
```

Comment:
This needs to land to mainline (`develop`) and we need to check out that one before we can merge the devcontainer